### PR TITLE
Add new course MigraCode Web Development

### PR DESCRIPTION
**Pull request by actuallyyun**
## What does this PR do?
Add resource(s)

## For resources
MigraCode Full Stack Web Development Course

### Why is this valuable (or not)?
It's comprehensive and interesting with loads of good exercises. It's entirely free and open to everyone.

### How do we know it's really free?
It's stated on the webpage. 

### For book lists, is it a book? For course lists, is it a course? etc.
A course

## Checklist:
- [ ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ ] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ ] Include author(s) and platform where appropriate.
- [ ] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [ ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -844,6 +844,8 @@
 * [ReactJS Tutorial for Beginners](https://www.youtube.com/playlist?list=PLC3y8-rFHvwgg3vaYJgHGnModB54rxOk3) - Codevolution
 * [Start Using React to Build Web Applications](https://egghead.io/courses/react-fundamentals) - Joe Maddalone
 * [The Beginner's Guide to React](https://egghead.io/courses/the-beginner-s-guide-to-react) - Kent C. Dodds
+* [React](https://syllabus.migracode.org/courses/introduction-3/course-content/react) - MigraCode Barcelona
```
Comment (shrn01) [inline courses/free-courses-en.md:847]: Looks like this is just a summary. Would it be a value add?

Comment (actuallyyun) [reply courses/free-courses-en.md:847]: It's interesting that this looks like a summary to you. It's the index page of the React module where you could click and enter the 3+1 React syllabus. ( 3 courses + 1 extra material)

Comment (eshellman) [reply courses/free-courses-en.md:847]: It's really more like the thrid lesson of the "Web Development Course". will move it out of React.

**Comment by eshellman** [comment]
Added in #9203. 
Thanks, and welcome to free-programming-books!

**Pull request closed without merging**